jenkins/Jenkinsfile

```diff
@@ -0,0 +1,240 @@
+import hudson.model.*
+import hudson.FilePath
+import jenkins.model.Jenkins
+
+//branch name we will create in leo. Will be appended with the build number.
+def branchName = "automated-leonardo-hash-update-"
+
+//used to exit early if no changes are detected
+def shouldAbortBuild = false
+
+//declared in outer scope so it persists between steps
+//omitting the 'def' is what makes it global
+LinkedHashMap<String, String> versionMap = [:]
+
+def ArrayList<String> getChangedImages() {
+  def String[] images = readFile("jenkins/imageDirs.txt").trim().split("\n")
```

Comment:
I like this to be something explicit personally, I think it's a better build practice to have the logic/detection be as minimal as possible, and tell it what to do through configs/files in the repo. It would already be pretty messy since we'd have to exclude a 3 different dirs, and if anyone adds any dirs that aren't images it would cause the job to fail.

I'd rather the failure case be it doesn't build an image because we didn't update a config rather than the job randomly starts failing because of an implementation detailed squirreled away in the job logic.

Also, it allows us flexibility in the sense that, if for some reason, we want to lock a version for a while or only detect changes on a specific image, we can just update the config.